### PR TITLE
Added custom layout support in NotFoundDelegate

### PR DESCRIPTION
**Pull request by mano87**
This is a better implementation for chaning the layout of error pages. See also the issue: #360 

I hope my pull request will be accepted :-)

### CODE REVIEW

File: src/Delegate/NotFoundDelegate.php [hunk 1]
```diff
@@ -93,9 +102,14 @@ private function generatePlainTextResponse(ServerRequestInterface $request)
...
     {
+        $templateData = [
+            'request'  => $request,
+            'layout'  => $this->layout,
+        ];
+
         $response = $this->responsePrototype->withStatus(StatusCodeInterface::STATUS_NOT_FOUND);
         $response->getBody()->write(
-            $this->renderer->render($this->template, ['request' => $request])
+            $this->renderer->render($this->template, $templateData)
```
Comment (xtreamwayz) [inline src/Delegate/NotFoundDelegate.php]: Why add a new variable?

```php
$this->renderer->render($this->template, [
    'request'  => $request,
    'layout'  => $this->layout,
]);
```

Comment (mano87) [reply src/Delegate/NotFoundDelegate.php]: I will rework this point to make the parameter optional - or do you have a better idea? Another option would be that in the factory other optional parameters can be passed to the renderer no matter what...

File: src/Delegate/NotFoundDelegate.php [hunk 2]
```diff
@@ -58,7 +67,7 @@ public function __construct(
      */
     public function process(ServerRequestInterface $request)
     {
-        if (! $this->renderer) {
+        if (!$this->renderer) {
```
Comment (webimpress) [inline src/Delegate/NotFoundDelegate.php]: Revert this change please. CS is now failing:
```
There must be a single space after a NOT operator; 0 found
```

File: src/Delegate/NotFoundDelegate.php [hunk 3]
```diff
@@ -78,7 +87,7 @@ private function generatePlainTextResponse(ServerRequestInterface $request)
             ->write(sprintf(
                 'Cannot %s %s',
                 $request->getMethod(),
-                (string) $request->getUri()
+                (string)$request->getUri()
```
Comment (webimpress) [inline src/Delegate/NotFoundDelegate.php]: Revert this change please. We don;t have currently sniff to check it, so all seems to be fine, but for consistency we should have here single space.

File: test/Delegate/NotFoundDelegateTest.php [hunk 4]
```diff
@@ -68,8 +70,10 @@ public function testUsesRendererToGenerateResponseContentsWhenPresent()
         $request = $this->prophesize(ServerRequestInterface::class)->reveal();
 
         $renderer = $this->prophesize(TemplateRendererInterface::class);
-        $renderer->render(NotFoundDelegate::TEMPLATE_DEFAULT, ['request' => $request])
-            ->willReturn('CONTENT');
+        $renderer->render(NotFoundDelegate::TEMPLATE_DEFAULT, [
+            'request' => $request,
+            'layout' => NotFoundDelegate::LAYOUT_DEFAULT
+            ])->willReturn('CONTENT');
```
Comment (webimpress) [inline test/Delegate/NotFoundDelegateTest.php]: The last line should have one less indent:
```php
        $renderer->render(NotFoundDelegate::TEMPLATE_DEFAULT, [
            'request' => $request,
            'layout' => NotFoundDelegate::LAYOUT_DEFAULT
        ])->willReturn('CONTENT');
```

File: test/Delegate/NotFoundDelegateTest.php [hunk 5]
```diff
@@ -68,8 +70,10 @@ public function testUsesRendererToGenerateResponseContentsWhenPresent()
         $request = $this->prophesize(ServerRequestInterface::class)->reveal();
 
         $renderer = $this->prophesize(TemplateRendererInterface::class);
-        $renderer->render(NotFoundDelegate::TEMPLATE_DEFAULT, ['request' => $request])
-            ->willReturn('CONTENT');
+        $renderer->render(NotFoundDelegate::TEMPLATE_DEFAULT, [
+            'request' => $request,
+            'layout' => NotFoundDelegate::LAYOUT_DEFAULT
```
Comment (webimpress) [inline test/Delegate/NotFoundDelegateTest.php]: Please add trailing comma after last element in the array

File: src/Delegate/NotFoundDelegate.php [hunk 6]
```diff
@@ -16,6 +16,7 @@
 class NotFoundDelegate implements DelegateInterface
 {
     const TEMPLATE_DEFAULT = 'error::404';
+    const LAYOUT_DEFAULT = 'layout::error';
```
Comment (weierophinney) [inline src/Delegate/NotFoundDelegate.php]: The main problem I have with this is that we do not at this time define a `layout::error` template in the skeleton application, only a `layout::default`. As such, I think that should likely be the default in this situation, to allow for backwards compatibility with existing applications.

**Comment by weierophinney** [comment]
Thanks, @mano87
